sql2.txt

```diff
@@ -17,29 +27,45 @@ may have ordered the same product in several different orders.  So the total qua
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
 
-
+SELECT CustomerName, ProductName, SUM(od.Quantity) AS TotalQuantity 
+FROM Products AS p 
+JOIN orderDetails AS od ON p.ProductID=od.ProductID 
+JOIN orders AS o  ON o.orderID=od.orderID 
+JOIN Customers AS c on c.CustomerID = o.CustomerID 
+GROUP BY c.CustomerID, p.ProductID;
 
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
 
+SELECT CustomerName, ProductName, SUM(od.Quantity) AS TotalQuantity, (SUM(Quantity) * Price) AS TotalPrice
+FROM Products AS p 
+JOIN orderDetails AS od ON p.ProductID=od.ProductID 
+JOIN orders AS o  ON o.orderID=od.orderID 
+JOIN Customers AS c on c.CustomerID = o.CustomerID 
+GROUP BY c.CustomerID, p.ProductID;
 
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
 
+INSERT INTO Orders ("CustomerID", "EmployeeID", "OrderDate", "ShipperID") 
+VALUES (90, 5, "7/9/1996 ", 2);
 
 6. Add three OrderDetails.  These should specify the ID of the new Order you just created.  For each,
 you will have to specify the ID of an existing product.  Paste your SQL statement below.
 
+INSERT INTO OrderDetails ("OrderID", "ProductID", "Quantity") 
+VALUES (10445, 5, 8), (10444, 8, 2), (10443, 1, 1);
 
 7.  Delete the third of the OrderDetails you just added.  Paste your SQL statement below.
 
-
+DELETE FROM OrderDetails WHERE OrderDetailID=519;
```

Comment:
The last OrderDetailID in `starter.db` is 518, so 519 should be the first of the OrderDetails you added, not the third.

```suggestion
DELETE FROM OrderDetails WHERE OrderDetailID=521;
```

Reply:
Done)